ui/app/routes/vault/cluster/clients/history.js

```diff
@@ -18,15 +19,20 @@ export default class HistoryRoute extends Route {
     try {
       let license = await this.store.queryRecord('license', {});
       // if license.startTime is 'undefined' return 'null' for consistency
-      return license.startTime || null;
+      return license.startTime || getStorage().getItem(INPUTTED_START_DATE) || null;
     } catch (e) {
-      // if error due to permission denied, return null so user can input date manually
-      return null;
+      // return null so user can input date manually
+      // if already inputted manually, will be in localStorage
+      return getStorage().getItem(INPUTTED_START_DATE) || null;
```

Comment:
Good question! It may not be that realistic in production, but while developing I had a license, but `license.startTime` was undefined, so wanted to check `localStorage` before returning `null`. And if `queryRecord` fails for some reason, we hit the `catch` and so wanted the same order of operations in the error block.